dtfs-central-api/src/v1/validation/utilisation-report-service/utilisation-report-validator.js

```diff
@@ -17,33 +17,68 @@ const {
 /**
  * Validates the month of the utilisation report. Returns null if valid, otherwise returns an error message.
  * @param {unknown} month - Month of utilisation report.
+ * @param {string} propertyName - String used in errors.
  * @returns {string | null} - Error message or null if valid.
  */
-const validateMonth = (month) => {
+const validateMonth = (month, propertyName) => {
   if (!month) {
-    return 'Month is required';
+    return `${propertyName} is required`;
   }
   if (!REGEXES.INTEGER_REGEX.test(month) || month < 1 || month > 12) {
-    return 'Month must be between 1 and 12';
+    return `${propertyName} must be between 1 and 12`;
   }
   return null;
 };
 
 /**
  * Validates the year of the utilisation report. Returns null if valid, otherwise returns an error message.
  * @param {unknown} year - year of utilisation report.
+ * @param {string} propertyName - String used in errors.
  * @returns {string | null} - Error message or null if valid.
  */
-const validateYear = (year) => {
+const validateYear = (year, propertyName) => {
   if (!year) {
-    return 'Year is required';
+    return `${propertyName} is required`;
   }
   if (!REGEXES.INTEGER_REGEX.test(year) || year < 2020 || year > 2100) {
-    return 'Year must be between 2020 and 2100';
+    return `${propertyName} must be between 2020 and 2100`;
   }
   return null;
 };
 
+/**
+ * Validates the report period for the utilisation report
+ * @param {unknown} reportPeriod - details of the report period.
+ * @returns {string[]} - an array of errors or an empty array if valid.
+ */
+const validateReportPeriod = (reportPeriod) => {
+  if (!reportPeriod) {
+    return ['Report period is required'];
+  }
+
+  const reportPeriodErrors = [];
+  const { month: startMonth, year: startYear,  } = reportPeriod.start;
+  const { month: endMonth, year: endYear } = reportPeriod.end;
+  const validateStartMonth = validateMonth(startMonth, 'startMonth');
+  if (validateStartMonth !== null) {
+    reportPeriodErrors.push(validateStartMonth)
+  }
```

Comment:
Since the response of `validateMonth` is an error, I think the following might make this a bit clearer:
```
const startMonthError = validateMonth(startMonth, 'startMonth');
if (startMonthError) {
  reportPeriodErrors.push(startMonthError)
}
```

Reply:
Modified this and all the similar values below